tests/gdb-tests/tests/heap/test_vis_heap_chunks.py

```diff
@@ -69,6 +69,16 @@ def vis_heap_line(heap_iter_offset=0x10, suffix=""):
     expected.append("%#x\t0x0000000000000000" % heap_iter())
     assert result == expected
 
+    ## This time using `default-visualize-chunk-number` to set `count`, to make sure that the config can work
+    gdb.execute("set default-visualize-chunk-number 1")
+    assert pwndbg.gdblib.config.default_visualize_chunk_number == 1
+    result = gdb.execute("vis_heap_chunk", to_string=True).splitlines()
+    assert result == expected
```

Comment:
Shouldn't expected be smaller here or sth?

Reply:
Sorry I didn’t catch that.
Do you mean to test a smaller default value here?

Reply:
I mean that we are asserting that result is the same as expected that was used for previous test/check but now we changed the default to 1 and it is a bit weird the expected still stays the same?

Reply:
I mean I would imagine that both result and expected should be smaller here now and only contain the 1 chunk now

Reply:
> I mean that we are asserting that result is the same as expected that was used for previous test/check but now we changed the default to 1 and it is a bit weird the expected still stays the same?

> I mean I would imagine that both result and expected should be smaller here now and only contain the 1 chunk now

The `result` and the `expected` should be the same here because before the line we added the test, we generated the `expected` to verify the output of `vis_heap_chunks 1`(which is the content of the `result`).
So both `result` and `expected` only contains 1 chunk here.

But yeah, I think the test here might be a little bit confusing it looks.
How about we test the config at the beginning like this:
```diff
diff --git a/tests/gdb-tests/tests/heap/test_vis_heap_chunks.py b/tests/gdb-tests/tests/heap/test_vis_heap_chunks.py
index edf602c..60bf6da 100644
--- a/tests/gdb-tests/tests/heap/test_vis_heap_chunks.py
+++ b/tests/gdb-tests/tests/heap/test_vis_heap_chunks.py
@@ -20,6 +20,21 @@ def test_vis_heap_chunk_command(start_binary):
     # Just a sanity check...
     assert (heap_page.start & 0xFFF) == 0

+    # Make sure `set default-visualize-chunk-number <num>` works properly
+    # Note: We only have 3 chunks(not including the top chunk) in the heap currently, so we only generate 3 outputs
+    for i in range(1, 4):
+        gdb.execute(f"set default-visualize-chunk-number {i}")
+        # `vis_heap_chunks` should show `i` chunks
+        assert gdb.execute("vis_heap_chunks", to_string=True) == gdb.execute(
+            f"vis_heap_chunks {i}", to_string=True
+        )
+    # Reset to the default config
+    gdb.execute(
+        f"set default-visualize-chunk-number {pwndbg.gdblib.config.default_visualize_chunk_number.default}"
+    )
+
```

Will it be clearer?

Reply:
I think its fine as it is; if u want to extend it, pls do in another PR